### PR TITLE
Add tags and description to vhost

### PR DESCRIPTION
**Pull request by kickster97**
### WHAT is this pull request doing?
add tags and description to vhost, and add them to details tuple. 
needed for client [compatibility](https://www.rabbitmq.com/docs/vhosts#:~:text=A%20description,set%20of%20tags)

discuss: do we want this? 

### HOW can this pull request be tested?
`GET /vhosts` response should have description and tags

### CODE REVIEW

**Review by snichme — APPROVED** [review]

**Review by spuun — COMMENTED** [review]
When retrieving a vhost via http api in rabbitmq the root object contains (among others) `description (string)`,  `tags (array)` and `metadata (object)`. The metadata object contains `description (string)` and  `tags (array)`. 

Dunno why it has that metadata object and values in the root object.

Anyway, based on this I'd say we should store tags as an array.

File: src/lavinmq/http/controller/vhosts.cr [hunk 1]
```diff
@@ -33,11 +33,14 @@ module LavinMQ
           u = user(context)
           refuse_unless_administrator(context, u)
           name = URI.decode_www_form(params["name"])
+          body = parse_body(context)
+          tags = body["tags"]?.to_s.split(",") || [] of String
```
Comment (spuun) [inline src/lavinmq/http/controller/vhosts.cr:37]: I think we should remove white spaces
```suggestion
          tags = body["tags"]?.to_s.split(",").map(&:strip) || [] of String
```

Maybe also remove empty strings?

I noticed that in rabbitmq tag string `a,b,, c,d,` results in tag array `["a","b","","c","d",""]`, which I don't think need to mimic.

Comment (carlhoerberg) [reply src/lavinmq/http/controller/vhosts.cr:37]:
```suggestion
          tags = body["tags"]?.to_s.split(',').map(&.strip).reject(&.empty?)
```

File: src/lavinmq/http/controller/vhosts.cr [hunk 2]
```diff
@@ -33,11 +33,14 @@ module LavinMQ
...
+          description = body["description"]?.try(&.to_s) || ""
```
Comment (carlhoerberg) [inline src/lavinmq/http/controller/vhosts.cr:38]: `nil.to_s == ""`
```suggestion
          description = body["description"]?.to_s
```

File: src/lavinmq/vhost.cr [hunk 3]
```diff
@@ -42,7 +42,8 @@ module LavinMQ
     @definitions_file_path : String
     @definitions_deletes = 0
 
-    def initialize(@name : String, @server_data_dir : String, @users : UserStore, @replicator : Replication::Replicator)
+    def initialize(@name : String, @tags : Array(String), @description : String,
+                   @server_data_dir : String, @users : UserStore, @replicator : Replication::Replicator)
```
Comment (carlhoerberg) [inline src/lavinmq/vhost.cr:46]:
```suggestion
    def initialize(@name : String, @server_data_dir : String, @users : UserStore, @replicator : Replication::Replicator, @description = "", @tags = Array(String).new(0))
```

File: spec/vhost_spec.cr [hunk 4]
```diff
@@ -99,7 +99,7 @@ describe LavinMQ::VHost do
       username = "test-user"
       user = Server.users.create(username, "password", [LavinMQ::Tag::Administrator])
       vhost = "test-vhost"
-      Server.vhosts.create(vhost, user)
+      Server.vhosts.create(vhost, [""], "", user)
```
Comment (carlhoerberg) [inline spec/vhost_spec.cr:102]: make tags and description optional

Comment (carlhoerberg) [reply spec/vhost_spec.cr:102]:
```suggestion
      Server.vhosts.create(vhost, user)
```

File: src/lavinmq/vhost_store.cr [hunk 5]
```diff
@@ -20,11 +20,11 @@ module LavinMQ
       end
     end
 
-    def create(name : String, user : User = @users.default_user, save : Bool = true)
+    def create(name : String, tags : Array(String) = [] of String, description : String = "", user : User = @users.default_user, save : Bool = true)
```
Comment (carlhoerberg) [inline src/lavinmq/vhost_store.cr:23]:
```suggestion
    def create(name : String, user : User = @users.default_user, description = "", tags = Array(String).new(0), save : Bool = true)
```

**Review by carlhoerberg — APPROVED** [review]

**Pull request merged**